render-test/runner.cpp

```diff
@@ -269,26 +260,31 @@ bool TestRunner::checkRenderTestResults(mbgl::PremultipliedImage&& actualImage,
             return false;
         }
     }
+#if !defined(SANITIZE)
     // Check memory metrics.
     for (const auto& expected : metadata.expectedMetrics.memory) {
         auto actual = metadata.metrics.memory.find(expected.first);
         if (actual == metadata.metrics.memory.end()) {
             metadata.errorMessage = "Failed to find memory probe: " + expected.first;
             return false;
         }
-        if (actual->second.peak > expected.second.peak) {
+        bool passed{false};
+        float delta{0.0f};
+        std::tie(passed, delta) = MemoryProbe::checkPeak(expected.second, actual->second);
+        if (!passed) {
             std::stringstream ss;
-            ss << "Allocated memory peak size at probe \"" << expected.first << "\" is "
-               << actual->second.peak << " bytes, expected is " << expected.second.peak << " bytes.";
+            ss << "Allocated memory peak size at probe \"" << expected.first << "\" is " << actual->second.peak
+               << " bytes, expected is " << expected.second.peak << "±" << delta << " bytes.";
 
             metadata.errorMessage = ss.str();
             return false;
```

Comment:
I think it would be useful to check both probes and report failures for each. Right now, 'peak' will shadow 'allocations' failure.